src/Bundle/DependencyInjection/SyliusResourceExtension.php

```diff
@@ -55,6 +59,18 @@ public function load(array $configs, ContainerBuilder $container): void
         $this->loadPersistence($config['drivers'], $config['resources'], $loader);
         $this->loadResources($config['resources'], $container);
 
+        $container->registerForAutoconfiguration(ProviderInterface::class)
+            ->addTag('sylius.state_provider')
```

Comment:
To make it work, we need to define these defaultIndexMethods on interface commented here: https://github.com/Sylius/SyliusResourceBundle/pull/595#discussion_r1124994312